pkg/controller/servicebinding/servicebinder.go

```diff
@@ -298,7 +297,15 @@ func (b *serviceBinder) bind() (reconcile.Result, error) {
 
 	b.logger.Debug("Saving data on intermediary secret...")
 
-	secretObj, err := b.secret.createOrUpdate(b.envVars, b.sbr.AsOwnerReference())
+	secret := newSecret(
+		b.dynClient,
+		b.sbr.GetNamespace(),
+		b.sbr.GetName(),
+		b.envVars,
+		b.sbr.AsOwnerReference(),
+	)
+
+	secretObj, err := secret.overwrite()
```

Comment:
This is fine for now, but can you see that newSecret and overwrite are being called only once and one after the other? This corroborates with the idea of refactor this further to a function with the parameters defined by newSecret and the output defined by overwrite.

Reply:
>  the idea of refactor this further to a function with the parameters defined by newSecret and the output defined by overwrite.
Okay. Let me do that.

Reply:
Done